library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java

```diff
@@ -373,20 +376,41 @@ public void seekTo(final int msec) {
             case STATE_PREPARED:
             case STATE_PAUSED:
             case STATE_PLAYBACK_COMPLETED:
-                mLock.lock();
                 if (mTrack == null) {
                     return;
                 }
                 mTrack.flush();
-                mExtractor.seekTo(((long) msec * 1000), MediaExtractor.SEEK_TO_PREVIOUS_SYNC);
-                Log.d(TAG, "seek completed, position: " + (mExtractor.getSampleTime() / 1000L));
-                if (owningMediaPlayer.onSeekCompleteListener != null) {
-                    owningMediaPlayer.onSeekCompleteListener.onSeekComplete(owningMediaPlayer);
-                }
-                mLock.unlock();
-                if (playing) {
-                    start();
-                }
+
+                final boolean wasPlaying = playing;
+
+                Thread t = new Thread(new Runnable() {
```

Comment:
I have followed the same pattern already in use in the "initStream" method.

My goal is to allow the user to quick switch between episodes, even when the episode is loading or seeking. The problem happens when the episodes weren't downloaded, and the queue is filled with streaming episodes.

Please take a look at this video. That's not a great user experience. No offense. :)

https://www.dropbox.com/s/f7nf0g2tpueqa0a/V%C3%ADdeo%2005-04-16%2015%2050%2004.mov?dl=0

After these code changes, I was able to do this, in my podcast's player:
https://www.dropbox.com/s/su72c0g3nrh5gyw/V%C3%ADdeo%2005-04-16%2015%2054%2001.mov?dl=0

It seems that we won't have problems with MediaExtractor because the next episode will use another instance.